p2p/discover/table.go

```diff
@@ -605,26 +609,37 @@ func (tab *Table) deleteInBucket(b *bucket, id enode.ID) *node {
 	return rep
 }
 
-// bumpInBucket updates the node record of n in the bucket.
-func (tab *Table) bumpInBucket(b *bucket, newRecord *enode.Node) bool {
+// bumpInBucket updates a node record if it exists in the bucket.
+func (tab *Table) bumpInBucket(b *bucket, newRecord *enode.Node, isInbound bool) (n *node, updated bool) {
 	i := slices.IndexFunc(b.entries, func(elem *node) bool {
 		return elem.ID() == newRecord.ID()
 	})
 	if i == -1 {
-		return false
+		return nil, false // node not in bucket
+	}
+
+	// Disallow updates unless the sequence number is increased.
+	// Note there is a special case for discv4: if the node contacts us (isInbound),
+	// it is allowed to update its own entry.
+	n = b.entries[i]
+	isUpdate := newRecord.Seq() > n.Seq()
+	isDiscv4Update := n.Seq() == 0 && newRecord.Seq() == 0 && isInbound
```

Comment:
For discv5 the message has already been authenticated by now, correct? So that means only the actual node could double sign sequence 0?

Is it an issue that now in discv5 inbound contacts can update their endpoint by replaying sequence 0?

Reply:
I don't think it's an issue. We could always be even stricter and check if it's an unsigned record. However, that seems excessive. We previously accepted all inbound updates, even ones with a lower sequence number, and in fact this is something I am still considering to add back. The node itself knows best.